src/test/java/httpserver/mocks/MockRequestParser.java

```diff
@@ -16,7 +16,7 @@
     @Override
     public Request parse(InputStream requestStream) {
         parseCalled = true;
-        return new Request();
+        return new Request("GET", "/simple_get","HTTP/1.1", null, null);
     }
```

Comment:
Instead of the string "GET" can you use the constant Methods.GET? It is good practice to find all the places where the strings are called (even/especially) in tests and update them.